plugins/services/src/js/reducers/serviceForm/Constraints.js

```diff
@@ -0,0 +1,125 @@
+  REMOVE_ITEM,
+  SET
+} from '../../../../../../src/js/constants/TransactionTypes';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+
+module.exports = {
+  JSONReducer(state, {type, path, value}) {
+    if (path == null) {
+      return state;
+    }
+
+    if (this.constraints == null) {
+      this.constraints = [];
+    }
+
+    const joinedPath = path.join('.');
+
+    if (joinedPath.search('constraints') !== -1) {
+      if (joinedPath === 'constraints') {
+        switch (type) {
+          case ADD_ITEM:
+            this.constraints.push({field: null, operator: null, value: null});
+            break;
+          case REMOVE_ITEM:
+            this.constraints = this.constraints.filter((item, index) => {
+              return index !== value;
+            });
+            break;
+        }
+
+        return this.constraints.filter((item) => {
+          return item.field != null && item.operator != null;
+        }).map(({field, operator, value}) => {
+          return value ? [field, operator, value] : [field, operator];
+        });
+      }
+
+      if (type !== SET) {
+        return state;
+      }
+
+      let index = joinedPath.match(/\d+/)[0];
+      let key = ['field', 'operator', 'value'].find(function (name) {
+        return `constraints.${index}.${name}` === joinedPath;
+      });
+      if (key !== -1) {
+        this.constraints[index][key] = value;
+      }
+    }
+
+    return this.constraints.filter((item) => {
+      return item.field != null && item.operator != null;
+    }).map(({field, operator, value}) => {
+      return value ? [field, operator, value] : [field, operator];
+    });
+  },
+
+  JSONParser(state) {
+    if (state.constraints == null) {
+      return [];
+    }
+
+    return state.constraints.reduce(function (memo, item, index) {
+      memo.push(new Transaction(['constraints'], index, ADD_ITEM));
+      memo.push(new Transaction([
+        'constraints',
+        index,
+        'field'
+      ], item[0], SET));
+      memo.push(new Transaction([
+        'constraints',
+        index,
+        'operator'
+      ], item[1], SET));
+      memo.push(new Transaction([
+        'constraints',
+        index,
+        'value'
+      ], item[2], SET));
+
+      return memo;
+    }, []);
+  },
+
+  FormReducer(state = [], {type, path, value}) {
+    if (path == null) {
+      return state;
+    }
+
+    let joinedPath = path.join('.');
+
+    if (joinedPath.search('constraints') !== -1) {
+      if (joinedPath === 'constraints') {
+        switch (type) {
+          case ADD_ITEM:
+            state.push({field: null, operator: null, value: null});
+            break;
+          case REMOVE_ITEM:
+            state = state.filter((item, index) => {
+              return index !== value;
+            });
+            break;
+        }
+
+        return state;
+      }
+
+      if (type !== SET) {
+        return state;
+      }
+
+      let index = joinedPath.match(/\d+/)[0];
```

Comment:
You always know that `index` is going to be the second item in the array, so you can do this: `const index = path[1];`